gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/PushConfig.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.pubsub;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * PubSub subscription push configuration.
+ */
+public final class PushConfig implements Serializable {
+
+  private static final long serialVersionUID = 4408885787064092231L;
+
+  private final String endpoint;
+  private final ImmutableMap<String, String> attributes;
+
+  public static final class Builder {
+
+    private String endpoint;
+    private final Map<String, String> attributes = new HashMap<>();
+
+    private Builder() {
+    }
+
+    public Builder endPoint(String endpoint) {
+      this.endpoint = checkNotNull(endpoint);
+      return this;
+    }
+
+    public Builder addAttribute(String name, String value) {
```

Comment:
https://cloud.google.com/pubsub/reference/rpc/google.pubsub.v1#google.pubsub.v1.PushConfig I will definitely reference it once I add javadoc.